core/src/main/java/software/amazon/awssdk/pagination/PaginatedItemsIterable.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.pagination;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.function.Function;
+
+public class PaginatedItemsIterable<ResponseT, ItemT> implements SdkIterable<ItemT> {
+
+    private final Iterator<ResponseT> responseIterator;
```

Comment:
Should these be iterables too? Otherwise it seems you can only iterate it once.

Reply:
Everytime this object is constructed, a new object of iterator is passed. So if customer tries to use the API to iterate through response/paginated member multiple items, it works fine. I have added a sample paginator class in the description.

Reply:
Hmm I'll have to see how this is used in the generated code but I assume it goes something like this.

```java
// I assume this creates the PaginatedItemsIterable
SdkIterable<Reservation> reservations = ec2.describeInstances().allItems();  

// Iterate once
for(Reservation r : reserverations) {...}

// What happens when I iterate again?
for(Reservation r : reserverations) {...}
```

Reply:
Good catch. As discussed offline, will expose a PaginatedResponses**Iterator** instead of the PaginatedResponses**Iterable** and pass the paginator ref to this class